app/uk/gov/hmrc/cgtpropertydisposalsfrontend/repos/NameMatchRetryStore.scala

```diff
@@ -45,22 +46,26 @@ trait NameMatchRetryStore {
 
 @Singleton
 class NameMatchRetryStoreImpl @Inject() (
-  mongo: ReactiveMongoComponent,
-  configuration: Configuration
+  mongo: MongoComponent,
+  configuration: Configuration,
+  timeStampSupport: TimestampSupport
 )(implicit
   ec: ExecutionContext
 ) extends NameMatchRetryStore
     with Repo {
 
-  val cacheRepository: CacheMongoRepository = {
+  val cacheRepository = {
     val expireAfter: FiniteDuration = configuration.underlying
       .get[FiniteDuration]("bpr-name-match.store.expiry-time")
       .value
 
-    new CacheMongoRepository("bpr-name-match-retries", expireAfter.toSeconds)(
-      mongo.mongoConnector.db,
-      ec
-    )
+    new MongoCacheRepository[String](
+      mongo,
+      "bpr-name-match-retries",
+      ttl = Duration.fromNanos(expireAfter.toSeconds),
+      timestampSupport = timeStampSupport,
+      cacheIdType = CacheIdType.SimpleCacheId
+    )(ec)
   }
 
   val sessionKey = "bpr-name-match-retries"
```

Comment:
NABD, but this val could have been declared before line no 62 and the can be reused at line no 64

Reply:
Close it if you are not implementing this.